internal/bminventory/inventory.go

```diff
@@ -1486,31 +1490,27 @@ func (b *bareMetalInventory) EnableHost(ctx context.Context, params installer.En
 	return installer.NewEnableHostOK().WithPayload(&host)
 }
 
-func (b *bareMetalInventory) DownloadClusterFiles(ctx context.Context, params installer.DownloadClusterFilesParams) middleware.Responder {
+func (b *bareMetalInventory) GetPresignedForClusterFiles(ctx context.Context, params installer.GetPresignedForClusterFilesParams) middleware.Responder {
 	log := logutil.FromContext(ctx, b.log)
-	var cluster common.Cluster
-	log.Infof("Download cluster files: %s for cluster %s", params.FileName, params.ClusterID)
-
-	if !funk.Contains(clusterFileNames, params.FileName) {
-		err := fmt.Errorf("invalid cluster file %s", params.FileName)
-		log.WithError(err).Errorf("failed download file: %s from cluster: %s", params.FileName, params.ClusterID)
-		return common.NewApiError(http.StatusBadRequest, err)
+	// Presigned URL only works with AWS S3 because scality is not exposed, in this case URL should be like s3.us-east-1.amazonaws.com
+	if !strings.HasSuffix(b.Config.S3EndpointURL, ".amazonaws.com") {
+		return common.NewApiError(http.StatusBadRequest, errors.New("Failed to generate presigned URL: invalid backend"))
 	}
-
-	if err := b.db.First(&cluster, "id = ?", params.ClusterID).Error; err != nil {
-		log.WithError(err).Errorf("failed to find cluster %s", params.ClusterID)
-		if gorm.IsRecordNotFoundError(err) {
-			return installer.NewDownloadClusterFilesNotFound().
-				WithPayload(common.GenerateError(http.StatusNotFound, err))
-		} else {
-			return installer.NewDownloadClusterFilesInternalServerError().
-				WithPayload(common.GenerateError(http.StatusInternalServerError, err))
-		}
+	if apiErr := b.checkFileForDownload(ctx, params.ClusterID.String(), params.FileName); apiErr != nil {
+		return apiErr
 	}
-	if err := b.clusterApi.DownloadFiles(&cluster); err != nil {
-		log.WithError(err).Errorf("failed to download cluster files %s", params.ClusterID)
-		return installer.NewDownloadClusterFilesConflict().
-			WithPayload(common.GenerateError(http.StatusConflict, err))
+	duration, _ := time.ParseDuration("10m")
```

Comment:
maybe we there is a point to have it configurable?

Reply:
Don't think so, they should download it right away